pkg/agent/manager/manager.go

```diff
@@ -97,12 +98,19 @@ func (m *manager) Initialize(ctx context.Context) error {
 
 	err := m.storePrivateKey(ctx, m.c.SVIDKey)
 	if err != nil {
-		return fmt.Errorf("fail to store private key: %v", err)
+		return fmt.Errorf("failed to store private key: %v", err)
 	}
 
 	m.backoff = backoff.NewBackoff(m.clk, m.c.SyncInterval)
 
-	return m.synchronize(ctx)
+	err = m.synchronize(ctx)
+	if nodeutil.IsShutdownError(err) {
+		m.c.Log.WithError(err).Error("Failed to synchronize: removing SVID")
+		if deleteErr := DeleteSVID(m.svidCachePath); deleteErr != nil {
+			m.c.Log.WithError(deleteErr).Error("Failed to remove SVID")
+		}
```

Comment:
we're doing this in a few spots. can this be moved to a shared function? (i.e. m.deleteSVID)